lib/puppet/parser/functions/new.rb

```diff
@@ -626,6 +626,21 @@
 | s         | Same as d.
 | p         | Same as d.
 
+### Binary value to String
+
+| Format    | Default formats
+| ------    | ---------------
+| s         | binary as unquoted UTF-8 characters (errors if byte sequence is invalid UTF-8)
+| p         | 'Binary("<base64strict>")'
+| b         | '<base64>' - base64 string with newlines inserted
+| B         | '<base64strict>' - base64 strict string (without newlines inserted)
+| u         | '<base64urlsafe>' - base64 urlsafe string
+| t         | 'Binary' - outputs the name of the type only
+| T         | 'BINARY' - output the name of the type in all caps only
+
+The alternate form flag `#` will quote the binary or base64 text output
```

Comment:
A '%#s' doesn't just quote the string. It also uses escaped bytes rather than as UTF-8. I think that deserves a mention.